### ISSUE TITLE
Node auto-healing unable to fully add new nodes

### ISSUE THREAD

**Issue by jessica-hofmeister** [vexxhost/magnum-cluster-api#173]
If you create a cluster via magnum and delete one of the instances (either a worker or a control-plane), a new instance will be created, however it will never fully be integrated into the cluster. See below for an example of what a cluster looks like after attempting to auto-heal

```
❯ clusterctl -n magnum-system describe cluster --color --grouping=false --show-conditions KubeadmControlPlane,MachineDeployment,Machine kube-5pow5
NAME                                                                 READY    SEVERITY  REASON                           SINCE  MESSAGE                                                                                                  
Cluster/kube-5pow5                                                   True                                                36m                                                                                                              
├─ClusterInfrastructure - OpenStackCluster/kube-5pow5-r7cxh                                                                                                                                                                               
├─ControlPlane - KubeadmControlPlane/kube-5pow5-82jk6                True                                                36m                                                                                                              
│ │           ├─Available                                            True                                                38m                                                                                                              
│ │           ├─CertificatesAvailable                                True                                                39m                                                                                                              
│ │           ├─ControlPlaneComponentsHealthy                        False    Error     ControlPlaneComponentsUnhealthy  28m    Following machines are reporting control plane errors: kube-5pow5-82jk6-xhnzc                             
│ │           ├─EtcdClusterHealthy                                   True                                                36m                                                                                                              
│ │           ├─MachinesCreated                                      True                                                36m                                                                                                              
│ │           ├─MachinesReady                                        True                                                36m                                                                                                              
│ │           └─Resized                                              True                                                36m                                                                                                              
│ ├─Machine/kube-5pow5-82jk6-7dhgn                                   True                                                39m                                                                                                              
│ │             ├─APIServerPodHealthy                                True                                                36m                                                                                                              
│ │             ├─BootstrapReady                                     True                                                39m                                                                                                              
│ │             ├─ControllerManagerPodHealthy                        True                                                36m                                                                                                              
│ │             ├─EtcdMemberHealthy                                  Unknown            MemberInspectionFailed           7m2s   Failed to connect to the etcd pod on the kube-5pow5-control-plane-ggm7k-cw8l5 node: could not establ ...  
│ │             ├─EtcdPodHealthy                                     True                                                37m                                                                                                              
│ │             ├─InfrastructureReady                                True                                                39m                                                                                                              
│ │             ├─NodeHealthy                                        True                                                37m                                                                                                              
│ │             └─SchedulerPodHealthy                                True                                                36m                                                                                                              
│ ├─Machine/kube-5pow5-82jk6-np4cq                                   True                                                37m                                                                                                              
│ │             ├─APIServerPodHealthy                                True                                                36m                                                                                                              
│ │             ├─BootstrapReady                                     True                                                37m                                                                                                              
│ │             ├─ControllerManagerPodHealthy                        True                                                36m                                                                                                              
│ │             ├─EtcdMemberHealthy                                  True                                                36m                                                                                                              
│ │             ├─EtcdPodHealthy                                     True                                                36m                                                                                                              
│ │             ├─InfrastructureReady                                True                                                37m                                                                                                              
│ │             ├─NodeHealthy                                        True                                                36m                                                                                                              
│ │             └─SchedulerPodHealthy                                True                                                36m                                                                                                              
│ └─Machine/kube-5pow5-82jk6-xhnzc                                   True                                                36m                                                                                                              
│               ├─APIServerPodHealthy                                False    Error     PodFailed                        28m    Missing node                                                                                              
│               ├─BootstrapReady                                     True                                                36m                                                                                                              
│               ├─ControllerManagerPodHealthy                        False    Error     PodFailed                        28m    Missing node                                                                                              
│               ├─EtcdMemberHealthy                                  Unknown            MemberInspectionFailed           28m    Failed to connect to the etcd pod on the kube-5pow5-control-plane-ggm7k-75qsv node: could not establ ...  
│               ├─EtcdPodHealthy                                     False    Error     PodFailed                        28m    Missing node                                                                                              
│               ├─InfrastructureReady                                True                                                36m                                                                                                              
│               ├─NodeHealthy                                        False    Error     NodeNotFound                     28m                                                                                                              
│               └─SchedulerPodHealthy                                False    Error     PodFailed                        28m    Missing node                                                                                              
└─Workers                                                                                                                                                                                                                                 
  └─MachineDeployment/kube-5pow5-default-worker-ktjkv                True                                                36m                                                                                                              
    │           └─Available                                          True                                                36m                                                                                                              
    ├─Machine/kube-5pow5-default-worker-ktjkv-56b9f4cc5xlkdsc-29cgk  True                                                37m                                                                                                              
    │             ├─BootstrapReady                                   True                                                37m                                                                                                              
    │             ├─InfrastructureReady                              True                                                37m                                                                                                              
    │             └─NodeHealthy                                      True                                                36m                                                                                                              
    ├─Machine/kube-5pow5-default-worker-ktjkv-56b9f4cc5xlkdsc-fvqdt  True                                                37m                                                                                                              
    │             ├─BootstrapReady                                   True                                                37m                                                                                                              
    │             ├─InfrastructureReady                              True                                                37m                                                                                                              
    │             └─NodeHealthy                                      True                                                36m                                                                                                              
    └─Machine/kube-5pow5-default-worker-ktjkv-56b9f4cc5xlkdsc-k9cpx  True                                                37m                                                                                                              
                  ├─BootstrapReady                                   True                                                37m                                                                                                              
                  ├─InfrastructureReady                              True                                                37m                                                                                                              
                  └─NodeHealthy                                      True                                                36m          
```

**Comment by mnaser** [comment]
I think the issue here is deleting a VM recreates it not cleanly..

Would it be possible to try and instead shutdown the VM and see if it cleans it up?  I think delete the VM messes things up in the CAPO.

**Comment by jessica-hofmeister** [comment]
Thanks for your suggestion, I tried powering down a worker node, which set it to a `not ready` state (as expected) and then powered it back on which cleanly set it back to a `ready` state (as expected).
However, if I power the node off, wait for it to go to a `not ready` state, and then delete the node, the end results are the same, where the new node that is created does not join the cluster successfully.

**Comment by mnaser** [comment]
@jessica-hofmeister have you tried doing a full shutdown and leaving the vm not started for 5 minutes and see if it starts a failover?

**Comment by mnaser** [comment]
@jessica-hofmeister I think you might be being hit by https://github.com/vexxhost/magnum-cluster-api/commit/700a19cc52bbf6a62983b3fbb6395eea5b2fa3b5 -- we haven't pushed out a new release that includes this, i think its a good time to push a new release.

**Comment by mnaser** [comment]
@jessica-hofmeister I just released 0.9.0 -- could you try and see if that resolves your issues?

**Comment by jessica-hofmeister** [comment]
We tested this morning and it was successful! I tested the below scenarios on worker nodes and control-plane nodes.

1) - turn vm off only, see what happens = node gets deleted automatically after a few minutes, new one shows up happy
2) - turn vm off, wait till it shows not ready, delete myself = node gone, new one shows up happy
3) - just delete vm manually = node gone, new one shows up happy! 

Thank you @mnaser!
